app/controllers/api/v1/statuses_controller.rb

```diff
@@ -202,6 +205,23 @@ def status_params
     )
   end
 
+  def quote_approval_policy
+    # TODO: handle `nil` separately
+    return nil unless Mastodon::Feature.outgoing_quotes_enabled? && status_params[:quote_approval_policy].present?
+
+    case status_params[:quote_approval_policy]
+    when 'public'
+      Status::QUOTE_APPROVAL_POLICY_FLAGS[:public] << 16
+    when 'followers'
+      Status::QUOTE_APPROVAL_POLICY_FLAGS[:followers] << 16
+    when 'nobody'
+      0
+    else
+      # TODO: raise more useful message
+      raise ActiveRecord::RecordInvalid
+    end
+  end
```

Comment:
No need to change this immediately, but do you think this could maybe go into the model?

I find this whole bit shifting to be really confusing and it is surprising to see in a controller. I think the more we hide this implementation detail away in the model (or rather the `Status::InteractionPolicyConcern`) the better.